tests/test_schema.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+
+"""Tests for the jsonschema node validation."""
+
+import unittest
+
+import pybel.dsl
+from pybel.schema import is_valid_node
+from pybel.testing.utils import n
+
+
+VALID_ENTITY = {
+    "namespace": "",
+    "name": "",
+    "identifier": ""
+}
+
+VALID_VARIANT = {
+    "kind": "",
+    "concept": VALID_ENTITY,
+    "code": "",
+    "pos": 0
+}
+
+VALID_ABUNDANCE = {
+    "function": "",
+    "concept": VALID_ENTITY,
+    "xrefs": [VALID_ENTITY, VALID_ENTITY],
+    "variants": [VALID_VARIANT, VALID_VARIANT]
+}
+
+NAMESPACE, NAME, IDENTIFIER = n(), n(), n()
+
+
+def build_abundance(variant_dict):
+    """Return an abundance dict with the given variant."""
+    abundance = VALID_ABUNDANCE.copy()
+    abundance["variants"] = [variant_dict, variant_dict]
+    return abundance
+
+
+class TestSchema(unittest.TestCase):
+    """Tests for the jsonschema node validation."""
+
+    def test_pure_abundances(self):
+        """Test validating abundance nodes."""
+        abundance = pybel.dsl.Abundance(namespace=NAMESPACE, name=NAME, identifier=IDENTIFIER)
+        protein = pybel.dsl.Protein(namespace=NAMESPACE, name=NAME,
+                                    identifier=IDENTIFIER)
+        self.assertTrue(is_valid_node(abundance))
+        self.assertTrue(is_valid_node(protein))
+
+    def test_variant_abundances(self):
+        """Test validating abundance nodes with variants."""
+        gene_mod = pybel.dsl.GeneModification(NAME, namespace=NAMESPACE, identifier=IDENTIFIER)
+        protein_sub = pybel.dsl.ProteinSubstitution('Abc', 1, 'Def')
+        protein_mod = pybel.dsl.ProteinModification(NAME, code='Abc', position=1,
```

Comment:
does it not thrown an error about using an invalid amino acid code?

Reply:
Added! Though it did take some nasty regex to check on `ProteinSubstitution`.